src/Router/types.ts

```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react';
+
+export interface RouterNavigateFunction {
+    (to: string): void;
+}
+
+export interface IRouter {
+    navigate: (path: string) => void;
+}
+
+export enum RouteName {
```

Comment:
Value of enum keys should match with values.

If you need to define routes, i would suggest to do to in implementation.

```ts
function getRoutePath(route: RouteName): string {
  switch(route) {
    case RouteName.LOGIN:
       return "/login";
    // ...
    default:
      return assertInvalidValue(route);
  }
}
```

Reply:
@Rastopyr 
But what I should do in case I need to put query or params? like video/:video_slag

Reply:
Not sure, that i understand the problem. Can you pls provide some scenario, where problem is highlighted?